src/app/ClusterInfo.h

```diff
@@ -23,6 +23,16 @@
 
 namespace chip {
 namespace app {
+
+/**
+ *
+ * @brief This is used to record the cluster's attribute path in server side read or subscription client interested in, for read interaction,
+ * read handler would mark those interested clusterInfo as dirty when receiving read request, then reporting engine would send out those
+ * dirty changes to client per interested clusterInfo, considering IM timed interaction, reporting engine would send out the dirty changes with delay.
+ * For subscription, in server side, if user modifies some of interested clusters, and those corresponding clusterInfo would be marked dirty,
+ * reporting engine would send them out.
```

Comment:
I can't make sense of most of this comment.  What is this object actually meant to represent?  If doing a wildcard read, will there be a whole bunch of these?  Or just one?

This object is also pretty wasteful size-wise (it uses 8 bytes to represent the boolean!), but that's secondary to the fact that it's really not clear what it is.

Reply:
we have not yet add wildcard support, assume readHandler receive one read request with wildcard, for example all clusters under particular endpoints,, reader handler would acquire those clusterInfos, mark them as dirty, reporting engine would send the attribute changes for those interested clustered.

Reply:
agree, 8 byte is wasteful here, maybe I should consider to unpack attributePathParam in clusterInfo, later I also need to add (EndpointId, ClusterId, EventId) and (EndpointId, ClusterId, CommandId).
ClusterInfo {
    NodeId mNodeId         = 0;
    ClusterId mClusterId   = 0;
    ListIndex mListIndex   = 0;
    EventId mEventId        = 0;
    EndpointId mEndpointId = 0;
    FieldId mFieldId       = 0;
    CommandId mCommandId = 0;
    bool mIsDirty = false;
}